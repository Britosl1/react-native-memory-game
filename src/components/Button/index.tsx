import React from 'react';
import {Text, View, StyleSheet, TouchableOpacity} from 'react-native';

interface IButtonProps {
  onPress: () => void;
  text: string;
}

const styles = StyleSheet.create({
  button: {
    borderRadius: 6,
    backgroundColor: '#663a82',
    height: 50,
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 50,
  },
  textStyle: {
    color: 'white',
    fontWeight: '800',
  },
});

const Button: React.FC<IButtonProps> = ({onPress, text}) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.button}>
        <Text style={styles.textStyle}>{text}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default Button;
