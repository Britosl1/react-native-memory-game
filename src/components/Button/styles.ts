import styled from 'styled-components/native';

export const ButtonContainer = styled.View`
  border-radius: 6px;
  background-color: #663a82;
  height: 50px;
  align-items: center;
  justify-content: center;
  margin-bottom: 50px;
`;

export const ButtonText = styled.Text`
  color: white;
  font-weight: 800;
`;
