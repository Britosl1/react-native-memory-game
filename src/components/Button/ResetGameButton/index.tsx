import React from 'react';
import {Image} from 'react-native';
import ResetIcon from '../../../images/reset-arrow.png';
import {ResetButtonContainer} from './styles';

interface IResetGameButtonProps {
  onPress: () => void;
}

const ResetGameButton: React.FC<IResetGameButtonProps> = ({onPress}) => {
  return (
    <ResetButtonContainer onPress={onPress}>
      <Image source={ResetIcon} />
    </ResetButtonContainer>
  );
};

export default ResetGameButton;
