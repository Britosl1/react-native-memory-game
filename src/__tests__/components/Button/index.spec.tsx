import React from 'react';
import {render} from '@testing-library/react-native';
import Button from '../../../components/Button';

describe('Snapshot test', () => {
  test('Button component', () => {
    const {toJSON} = render(<Button text="Clique aÃ­" onPress={jest.fn()} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
