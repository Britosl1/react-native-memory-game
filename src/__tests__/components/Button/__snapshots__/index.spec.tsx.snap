// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Button component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#663a82",
          "borderRadius": 6,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 50,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontWeight": "800",
          },
        ]
      }
    >
      Clique a√≠
    </Text>
  </View>
</View>
`;
